modules/control-utility/src/test/java/org/apache/ignite/util/GridCommandHandlerConsistencyOnClusterCrashTest.java

```diff
@@ -0,0 +1,151 @@
+            .setWalMode(WALMode.FSYNC); // Allows to use special IO at WAL as well.
+
+        cfg.setCacheConfiguration(new CacheConfiguration<>()
+            .setAffinity(new RendezvousAffinityFunction(false, 1))
+            .setBackups(backupNodes)
+            .setName(DEFAULT_CACHE_NAME)
+            .setAtomicityMode(TRANSACTIONAL)
+            .setWriteSynchronizationMode(FULL_SYNC) // Allows to be sure that all messages are sent when put succeed.
+            .setReadFromBackup(true)); // Allows to check values on backups for idle_verify.
+
+        return cfg;
+    }
+
+    /** */
+    @Test
+    public void testLwmCountersOnClusterCrash() throws Exception {
+        int prepareRqNum = 20;
+        int finishRqNum = 30;
+
+        IgniteEx ignite = startGrids(nodes);
+        ignite.cluster().state(ClusterState.ACTIVE);
+
+        final AtomicInteger updateCnt = new AtomicInteger();
+
+        // Enough to have historical rebalance when needed.
+        for (int i = 0; i < 2_000; i++)
+            ignite.cache(DEFAULT_CACHE_NAME).put(updateCnt.incrementAndGet(), 0);
+
+        stopAllGrids();
+        startGrids(nodes);
+
+        Ignite prim = primaryNode(0L, DEFAULT_CACHE_NAME);
+
+        CountDownLatch prepareLatch = new CountDownLatch(backupNodes * prepareRqNum);
+        CountDownLatch finishLatch = new CountDownLatch(backupNodes * finishRqNum);
+
+        TestRecordingCommunicationSpi.spi(prim).blockMessages(new IgniteBiPredicate<ClusterNode, Message>() {
+            @Override public boolean apply(ClusterNode node, Message msg) {
+                if (msg instanceof GridDhtTxPrepareRequest && prepareLatch.getCount() > 0) {
+                    prepareLatch.countDown();
+
+                    return true;
+                }
+                else if (msg instanceof GridDhtTxFinishRequest && finishLatch.getCount() > 0) {
+                    finishLatch.countDown();
+
+                    return true;
+                }
+                else
+                    return false;
+            }
+        });
+
+        IgniteCache<Integer, Integer> primCache = prim.cache(DEFAULT_CACHE_NAME);
+
+        for (int i = 0; i < prepareRqNum; i++)
+            GridTestUtils.runAsync(() -> primCache.put(updateCnt.incrementAndGet(), 0));
+
+        prepareLatch.await(getTestTimeout(), TimeUnit.MILLISECONDS);
+
+        for (int i = 0; i < finishRqNum; i++)
+            GridTestUtils.runAsync(() -> primCache.put(updateCnt.incrementAndGet(), 0));
+
+        finishLatch.await(getTestTimeout(), TimeUnit.MILLISECONDS);
+
+        G.allGrids().forEach(
+            node -> ((ThrowableFileIOTestFactory)node.configuration().getDataStorageConfiguration().getFileIOFactory())
+                .setThrowEx(true));
+
+        stopAllGrids();
+        startGrids(nodes);
+
+        awaitPartitionMapExchange();
+
+        assertEquals(EXIT_CODE_OK, execute("--cache", "idle_verify"));
+        assertConflicts(false, false);
```

Comment:
This code will be successful when all data missed too :) 
It also will be successful at full rebalance (current case) :(

Please check the result properly.